src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/HomeServiceTest.cs

```diff
@@ -13,145 +13,148 @@ namespace DevAdventCalendarCompetition.Tests
+            var result = homeService.GetTestAnswerByUserId(testAnswer.UserId, testAnswer.Id);
 
             // Assert
             Assert.IsType<TestAnswerDto>(result);
-            this._homeRepositoryMock.Verify(mock => mock.GetTestAnswerByUserId(It.Is<string>(x => x.Equals(this._testAnswer.UserId, StringComparison.Ordinal)), It.Is<int>(x => x.Equals(this._testAnswer.Id))), Times.Once());
         }
 
         [Fact]
         public void GetTestsWithUserAnswers_ReturnTestWithAnswerListDto()
         {
             // Arrange
-            this._homeRepositoryMock.Setup(mock => mock.GetTestsWithUserAnswers()).Returns(this._testList);
+            var testList = GetTestList();
+            this._testRepositoryMock.Setup(mock => mock.GetTestsWithUserAnswers()).Returns(testList);
             this._mapper = new MapperConfiguration(cfg => cfg.AddProfile<TestProfile>()).CreateMapper();
-            var homeService = new HomeService(this._homeRepositoryMock.Object, this._mapper);
+            var homeService = new HomeService(this._resultsRepositoryMock.Object, this._testAnswerRepositoryMock.Object, this._testRepositoryMock.Object, this._mapper);
 
             // Act
             var result = homeService.GetTestsWithUserAnswers();
 
             // Assert
             Assert.IsType<List<TestWithAnswerListDto>>(result);
-            Assert.True(this._testList.Count == result.Count);
-            this._homeRepositoryMock.Verify(mock => mock.GetTestsWithUserAnswers(), Times.Once());
+            Assert.True(testList.Count == result.Count);
         }
+
+        private static Test GetCurrentTest() => new Test()
+        {
+            Id = 2,
+            Number = 2,
+            StartDate = DateTime.Today.AddHours(12),
+            EndDate = DateTime.Today.AddHours(23).AddMinutes(59),
+            Answers = null
+        };
+
+        private static TestAnswer GetTestAnswer() => new TestAnswer()
+        {
+            Id = 1,
+            Test = new Test(),
+            TestId = 1,
+            User = new ApplicationUser(),
+            UserId = "1",
+            AnsweringTimeOffset = default,
+            AnsweringTime = DateTime.Now
+        };
+
+        private static List<Test> GetTestList() => new List<Test>()
+        {
+            new Test()
+            {
+                Id = 1,
+                Number = 1,
+                StartDate = DateTime.Today.AddHours(12),
+                EndDate = DateTime.Today.AddHours(23).AddMinutes(59),
+                Answers = null
+            }
+        };
+
+        private static Test GetOldTest() => new Test()
+        {
+            Id = 1,
+            Number = 1,
+            StartDate = DateTime.Today.AddDays(-2).AddHours(12),
+            EndDate = DateTime.Today.AddDays(-2).AddHours(23).AddMinutes(59),
+            Answers = null
+        };
+
+        private static Test GetFutureTest() => new Test()
+        {
```

Comment:
I suggest setting parameter `Number` based on current test number.
And what does it mean Future? Not started? The name should be more meaningful

Reply:
I have adhered to names, which were before the changes, but I agree that this might be confusing